### PR TITLE
Issue warning when accessing S3 without specifying anon=True

### PR DESCRIPTION
**Pull request by forman**
* xcube now issues a warning, if a data cube is opened from object 
  storage, and credentials have been neither passed not can be found, 
  and the object storage has been opened with the default `anon=False`. (#412)
* xcube no longer internally caches directory listings, which prevents 
  the situation where a data cube, that has recently been written into object 
  storage, cannot be found. 

This addresses issue #412 posted by @tiagoams.

Checklist:

* [ ] ~Add unit tests and/or doctests in docstrings~ (would require boto-core mock)
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] ~New/modified features documented in `docs/source/*`~
* [x] Changes documented in `docs/CHANGES.md`
* [x] AppVeyor and Travis CI passes
* [x] Test coverage remains or increases (target 100%)
* [ ] Associated issues closed after merge

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -12,6 +12,12 @@
   1. Generator API `xcube.core.gen2.LocalCubeGenerator` and
     `xcube.core.gen2.service.RemoteCubeGenerator`;
   2. Generator CLI `xcube gen2`.
+* xcube now issues a warning, if a data cube is opened from object 
+  storage, and credentials have been neither passed not can be found,
```
Comment (TonioF) [inline CHANGES.md:16]:
```suggestion
  storage, and credentials either were passed or can not be found, 
```

File: CHANGES.md [hunk 2]
```diff
@@ -12,6 +12,12 @@
   1. Generator API `xcube.core.gen2.LocalCubeGenerator` and
     `xcube.core.gen2.service.RemoteCubeGenerator`;
   2. Generator CLI `xcube gen2`.
+* xcube now issues a warning, if a data cube is opened from object 
+  storage, and credentials have been neither passed not can be found, 
+  and the object storage has been opened with the default `anon=False`. (#412)
+* xcube no longer internally caches directory listings, which prevents 
+  the situation where a data cube, that has recently been written into object 
+  storage, cannot be found.
```
Comment (TonioF) [inline CHANGES.md:20]:
```suggestion
  the situation where a data cube that has recently been written into object 
  storage cannot be found. 
```

File: xcube/core/dsio.py [hunk 3]
```diff
@@ -563,17 +564,79 @@ def get_path_or_s3_store(path_or_url: str,
...
+        # because we want current contents
+        s3_kwargs['use_listings_cache'] = False
+    client_kwargs = s3_kwargs.pop('client_kwargs', {})
+    client_kwargs.update(s3_client_kwargs or {})
+    try:
+        return s3fs.S3FileSystem(**s3_kwargs,
+                                 client_kwargs=client_kwargs)
+    except botocore.exceptions.NoCredentialsError:
+        if s3_kwargs.get('anon') is False:
+            warnings.warn('No object storage credentials where'
```
Comment (TonioF) [inline xcube/core/dsio.py:632]:
```suggestion
            warnings.warn('No object storage credentials were'
```

**Review by TonioF — CHANGES_REQUESTED** [review]
I get 21 test failures from test_s3. There seems to be a problem with the region name: TypeError: expected string or bytes-like object . Please have a look.

File: CHANGES.md (hunk 1 above)
Comment (forman) [reply CHANGES.md:16]: not --> nor is correct

Comment (TonioF) [reply CHANGES.md:16]: I f you want to use ''nor', say:
"credentials neither were passed nor can they be found" .
I don't like this solution as it sounds weird, so I still stand by my suggestion (I just noticed it is missing a "not").

**Review by TonioF — APPROVED** [review]
Looks good!

**Review by AliceBalfanz — APPROVED** [review]
Wonderful! Thank you :)

**Pull request merged**